[SPARK-28798][DOC][SQL]Document DROP TABLE/VIEW statement in SQL Reference.

shivusondur opened the pull request:
<!--
Thanks for sending a pull request!  Here are some tips for you:
  1. If this is your first time, please read our contributor guidelines: https://spark.apache.org/contributing.html
  2. Ensure you have added or run the appropriate tests for your PR: https://spark.apache.org/developer-tools.html
  3. If the PR is unfinished, add '[WIP]' in your PR title, e.g., '[WIP][SPARK-XXXX] Your PR title ...'.
  4. Be sure to keep the PR description updated to reflect all changes.
  5. Please write your PR title to summarize what this PR proposes.
  6. If possible, provide a concise example to reproduce the issue for a faster review.
-->

### What changes were proposed in this pull request?
Added doc for DROP TABLE and DROP VIEW sql command
<!--
Please clarify what changes you are proposing. The purpose of this section is to outline the changes and how this PR fixes the issue. 
If possible, please consider writing useful notes for better and faster reviews in your PR. See the examples below.
  1. If you refactor some codes with changing classes, showing the class hierarchy will help reviewers.
  2. If you fix some SQL features, you can provide some references of other DBMSes.
  3. If there is design documentation, please add the link.
  4. If there is a discussion in the mailing list, please add the link.
-->


### Why are the changes needed?
For reference DROP TABLE  or DROP VIEW in spark-sql
<!--
Please clarify why the changes are needed. For instance,
  1. If you propose a new API, clarify the use case for a new API.
  2. If you fix a bug, you can clarify why it is a bug.
-->


### Does this PR introduce any user-facing change?
It updates DROP TABLE  or DROP VIEW reference doc
<!--
If yes, please clarify the previous behavior and the change this PR proposes - provide the console output, description and/or an example to show the behavior difference if possible.
If no, write 'No'.
-->


### How was this patch tested?
<details>
<summary> Attached the Snap</summary>

DROP TABLE

![image](https://user-images.githubusercontent.com/7912929/67884038-2443b400-fb6b-11e9-9773-b21dae398789.png)
![image](https://user-images.githubusercontent.com/7912929/67797387-aa96c200-faa7-11e9-90d4-fa8b7c6a4ec7.png)

DROP VIEW
![image](https://user-images.githubusercontent.com/7912929/67797463-c306dc80-faa7-11e9-96ec-e2f2e89d0db8.png)
![image](https://user-images.githubusercontent.com/7912929/67797648-1ed16580-faa8-11e9-9d32-19106326e3d9.png)

</details>
<!--
If tests were added, say they were added here. Please make sure to add some test cases that check the changes thoroughly including negative and positive cases if possible.
If it was tested in a way different from regular unit tests, please clarify how you tested step by step, ideally copy and paste-able, so that other reviewers can test and check, and descendants can verify in the future.
If tests were not added, please describe why they were not added and/or why it was difficult to add.
-->

shivusondur committed:
Added doc

AmplabJenkins commented:
Can one of the admins verify this patch?

AmplabJenkins commented:
Can one of the admins verify this patch?

AmplabJenkins commented:
Can one of the admins verify this patch?

shivusondur committed:
ADDED THE VIEW DOC

dilipbiswal commented:
@shivusondur Could you please attach the screenshot of the page to make it easier for reviewers ?

srowen commented on docs/sql-ref-syntax-ddl-drop-table.md:
```diff
@@ -19,4 +19,20 @@ license: |
-**This page is under construction**
+### Description
+Drop a table and delete the directory associated with the table from the file system if this is not an
+EXTERNAL table. If the table to drop does not exist, an exception is thrown.
```
Back-tick the reserved words

shivusondur replied:
Handled

shivusondur committed:
Updated the document details

shivusondur committed:
Updated the document details

shivusondur commented:
DROP TABLE
![image](https://user-images.githubusercontent.com/7912929/63557319-d22c5400-c565-11e9-8f86-0d4afea20037.png)

DROP VIEW
![image](https://user-images.githubusercontent.com/7912929/63557340-ed975f00-c565-11e9-8ce1-05ce5b7bb97b.png)

shivusondur commented:
> @shivusondur Could you please attach the screenshot of the page to make it easier for reviewers ?

@dilipbiswal 
attached the screenshot

dilipbiswal commented:
@shivusondur A few related DOC PRs were merged by @gatorsmile 
https://github.com/apache/spark/pull/25525
https://github.com/apache/spark/pull/25523

Could you please make changes by following the format of these PRs ? A few things to note: 
1) Formatting of the parameters under `dl` tag now.
2) Examples should contain output. Please refer to the INSERT PR for examples.
3) Add some related statements - statements users are likely to visit. For example after a drop users may want to do a show tables to check the catalog or be interested in some other view commands.

shivusondur committed:
Updated the document according to latest comments

shivusondur committed:
Updated the document according to latest comments

shivusondur committed:
Updated the document according to latest comments

shivusondur commented:
@dilipbiswal @srowen @gatorsmile 
Updated the PR according to #25525 and #25523 PR's

Please review kindly check for merging.

dilipbiswal commented on docs/sql-ref-syntax-ddl-drop-table.md:
```diff
@@ -19,4 +19,70 @@ license: |
+### Description
+
+Deletes the table and removes the directory associated with this table from the file system
+if this is not `EXTERNAL` table. If the table is not present it throws exception.
```
Nit: an exception

dilipbiswal commented on docs/sql-ref-syntax-ddl-drop-table.md:
```diff
@@ -19,4 +19,70 @@ license: |
+Deletes the table and removes the directory associated with this table from the file system
+if this is not `EXTERNAL` table. If the table is not present it throws exception.
+
+In case of External table it only deletes the metadata from metastore database and it will not remove the directory 
```
Nit: In case of External table, only the associated metadata information are removed from the metastore database.

dilipbiswal commented on docs/sql-ref-syntax-ddl-drop-table.md:
```diff
@@ -19,4 +19,70 @@ license: |
+  </dd>
+  <dt><code><em>database_name</em></code></dt>
+  <dd>
+     Database name where table is present.
```
Nit: Specifies the database name where table is present.

dilipbiswal commented on docs/sql-ref-syntax-ddl-drop-table.md:
```diff
@@ -19,4 +19,70 @@ license: |
+  </dd>
+  <dt><code><em>table_name</em></code></dt>
+  <dd>
+     table name to be dropped.
```
Nit: Ditto.

dilipbiswal commented on docs/sql-ref-syntax-ddl-drop-table.md:
```diff
@@ -19,4 +19,70 @@ license: |
+-- Assumes a table name `employeetable` exist.
+DROP TABLE employeetable;
++---------+--+
+| Result  |
```
Can you please double check this ? The format looks a bit odd.

dilipbiswal commented on docs/sql-ref-syntax-ddl-drop-view.md:
```diff
@@ -19,4 +19,64 @@ license: |
 
-**This page is under construction**
+### Description
+Drops the specified `VIEW`, which was created by `CREATE VIEW` statement. `DROP VIEW` only involves
```
How about the following ?
DROP VIEW` removes the metadata associated with a specified view from the catalog.

dilipbiswal commented on docs/sql-ref-syntax-ddl-drop-view.md:
```diff
@@ -19,4 +19,64 @@ license: |
+  </dd>
+  <dt><code><em>database_name</em></code></dt>
+  <dd>
+     Database name where view is present.
```
Please add "Specifies"

AmplabJenkins commented:
Can one of the admins verify this patch?

shivusondur committed:
Handled the dilipbiswal's comments
Handled PR according to #25523 PR which is already merged.
to make uniform accross all docs

shivusondur commented:
@dilipbiswal 
Thanks for your time.
i handled all the comments and updated the screen shot.
Please check

gatorsmile commented:
ok to test

SparkQA commented:
**[Test build #111093 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/111093/testReport)** for PR 25533 at commit [`194a0b0`](https://github.com/apache/spark/commit/194a0b09a29c313be6180f29415d3360ab5fd86e).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/16177/
Test PASSed.

SparkQA commented:
**[Test build #111093 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/111093/testReport)** for PR 25533 at commit [`194a0b0`](https://github.com/apache/spark/commit/194a0b09a29c313be6180f29415d3360ab5fd86e).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/111093/
Test PASSed.

srowen commented on docs/sql-ref-syntax-ddl-drop-table.md:
```diff
@@ -19,4 +19,69 @@ license: |
-**This page is under construction**
+### Description
+
+Deletes the table and removes the directory associated with this table from the file system
```
Go ahead and say `DROP TABLE` deletes ...

srowen commented on docs/sql-ref-syntax-ddl-drop-table.md:
```diff
@@ -19,4 +19,69 @@ license: |
+Deletes the table and removes the directory associated with this table from the file system
+if this is not `EXTERNAL` table. If the table is not present it throws an exception.
+
+In case of External table, only the associated metadata information are removed from the metastore database.
```
of an external table

srowen commented on docs/sql-ref-syntax-ddl-drop-view.md:
```diff
@@ -19,4 +19,63 @@ license: |
 
-**This page is under construction**
+### Description
+DROP VIEW removes the metadata associated with a specified view from the catalog.
```
Back-tick DROP VIEW

srowen commented on docs/sql-ref-syntax-ddl-drop-view.md:
```diff
@@ -19,4 +19,63 @@ license: |
+
+### Example
+{% highlight sql %}
+-- Assumes a view name `employeeView` exist.
```
exists

srowen commented on docs/sql-ref-syntax-ddl-drop-view.md:
```diff
@@ -19,4 +19,63 @@ license: |
++---------+--+
++---------+--+
+
+-- Assumes a view name `employeeView` exist in `userdb` database
```
view named
in -> in the

srowen commented on docs/sql-ref-syntax-ddl-drop-view.md:
```diff
@@ -19,4 +19,63 @@ license: |
+{% endhighlight %}
+
+### Related Statements
+- [CREATE VIEW ](sql-ref-syntax-ddl-create-view.html)
```
Tiny nit: extra space inside bracket

shivusondur committed:
handled the comments

shivusondur commented:
@srowen 
Handled your comments, please check

SparkQA commented:
**[Test build #112863 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/112863/testReport)** for PR 25533 at commit [`df23c95`](https://github.com/apache/spark/commit/df23c952bbf27ce1a637370cf62d58526389473f).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/17793/
Test PASSed.

SparkQA commented:
**[Test build #112863 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/112863/testReport)** for PR 25533 at commit [`df23c95`](https://github.com/apache/spark/commit/df23c952bbf27ce1a637370cf62d58526389473f).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/112863/
Test PASSed.

srowen commented on docs/sql-ref-syntax-ddl-drop-table.md:
```diff
@@ -19,4 +19,69 @@ license: |
-**This page is under construction**
+### Description
+
+`DROP TABLE` deletes the table and removes the directory associated with this table from the file system
```
Nit: this -> the in both places

srowen reviewed (commented):
Looking fine, just a few more nits

shivusondur committed:
handled the comments

AmplabJenkins commented:
Merged build finished. Test PASSed.

shivusondur commented:
@srowen 
Thanks for reviewing.
I handled your comments.

SparkQA commented:
**[Test build #112950 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/112950/testReport)** for PR 25533 at commit [`2ca9a6e`](https://github.com/apache/spark/commit/2ca9a6e149c545d854c951c198c7a283c8ceb434).

SparkQA commented:
**[Test build #112950 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/112950/testReport)** for PR 25533 at commit [`2ca9a6e`](https://github.com/apache/spark/commit/2ca9a6e149c545d854c951c198c7a283c8ceb434).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

srowen commented:
Merged to master
